lib/IRGen/IRGenSIL.cpp

```diff
@@ -2239,7 +2239,7 @@ void IRGenSILFunction::emitSILFunction() {
 
   // Emit distributed accessor, and mark the thunk as accessible
   // by name at runtime through it.
-  if (CurSILFn->isDistributed() && CurSILFn->isThunk()) {
+  if (CurSILFn->isDistributed() && CurSILFn->isThunk() == IsThunk) {
```

Comment:
Whoa, I see...

Thank you so much for the fix 🙏

Reply:
Moving everything into AST enabled optimizations at call sites, which is nice.

Reply:
Yeah, very cool to see that! A bit of a pain to adjust everything but fantastic that we get those benefits <3